_posts/2015-09-xx-creating-systems-with-pipelining.md

```diff
@@ -0,0 +1,52 @@
+These templates are predefined with a set of CPU,RAM and Disc space and an OS specific mounted iPXE kickstart iso file. If it's needed, you can create and run an additional job changing the values of the machine before the install starts.
+A bit more challenging is to manage the network environment. In the simplest case a development system should be created, so the system needs an IP with internet access and an R/DNS entry.
+Here we killed two birds with one stone. By using i-doit we provide at first the information about the new system to our documentation system via REST API and in the second step calculate the next free ip in our network.
+Now we have the system documented and we have all information we need to create a [DHCP](https://en.wikipedia.org/wiki/Dynamic_Host_Configuration_Protocol) and [DNS](https://en.wikipedia.org/wiki/Domain_Name_System) entry. Using the API of i-doit the DHCP and DNS configuration files were generated, tested and deployed to our DHCP and DNS Servers.
+
+## Installation process
+
+Now we can start the virtual machine with our kickstart script installing the predefined OS. The kickstart script uses iPXE boot and loads the configuration and installs the system.
+The open source boot firmware iPXE allows us to boot an iso file over the network and allows dynamic or static IP configuration depending on your network. For example for networks without DHCP you can hard code a static ip for an installation and change the network configuration after the initial install is done. The loaded ISO files are preseeds for the specified os that include all information the install process needs to install the OS.
+At the moment Centos 6/7 and Debian 7/8 are supported from our side. After the installation, we have a first simple not very specialized system that could now be used. Now some magic Jenkins jobs do the trick and specialize the system.
+
+## Specialization
+
+From here on you can let your creativity run free. Here are many jobs available from [security related configurations](https://www.linode.com/docs/security/securing-your-server) over installing of different software to installing our full ePages to a system with a specific version or project specific configurations.
+In our situation it depends on the use case. For example internal development systems have other security related configurations then external systems, or
+test and live systems can have project specific configurations depending on what our custermers need.
+
+## Big Picture
+
+Here is a simplified picture of the whole pipeline. The full process lasts about 10 to 15 minutes but depends on the specific configuration you choose.
+
+{% image blog-pipeline-Jenkins.png %} {% endimage %}
+
+In the Future we want to use the existing build pipeline's to create distributed systems with our and other software.
```

Comment:
Thank you very much for this blog post! Great stuff and insights!
It would be cool if you could use line breaks after every sentence. It's easier to proof-read in GitHub :-)

When you're done with your changes, please pass it on here in GitHub to "steffimeier" and "maletzke". They will give their feedback. When this is done, pass it on to "RealMattMurdoch". Matt will do the final native check.